### PR TITLE
Add experimental aarch64-unknown-linux-gnu host

### PR DESCRIPTION
**Pull request by Hoverbear**
https://github.com/ferrocene/ferrocene/pull/595/files got messy, this re-approaches it from a modern `main`.

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try-

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

File: ferrocene/tools/self-test/src/targets.rs [hunk 1]
```diff
@@ -10,20 +10,28 @@ use crate::linkers::Linker;
...
+        linker: Linker::CrossCc(&["x86_64-linux-gnu-"]),
+    },
     TargetSpec { triple: "aarch64-unknown-none", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "armv8r-none-eabihf", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "thumbv7em-none-eabihf", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "thumbv7em-none-eabi", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "armv7r-none-eabihf", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "armebv7r-none-eabihf", std: false, linker: Linker::BundledLld },
-    TargetSpec { triple: "aarch64-unknown-nto-qnx710", std: true, linker: Linker::BundledLld },
-    TargetSpec { triple: "x86_64-pc-nto-qnx710", std: true, linker: Linker::BundledLld },
```
Comment (pietroalbini) [inline ferrocene/tools/self-test/src/targets.rs:26]: Why was QNX removed?

**Review by pietroalbini — APPROVED** [review]
bors merge

**Comment by Hoverbear** [comment]
I forgot a couple things. :( Should be fixed now.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Comment by Hoverbear** [comment]
bors try

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**